### ISSUE TITLE
--env doesn't work when --process process.json is used

### ISSUE THREAD

**Issue by kocho1984** [opencontainers/runc#3241]
Is it possible to use both --env and --process flags when running runc? I noticed that when --process is used (I have env variables defined in json file), env variables defined after --env are ignored (I don't see them inside a container). Is it expected behavior? I think it should be possible to define env variables both in json file and with --env option.

**Comment by JasonYangShadow** [comment]
I am not sure if my reproducibility is correct, I tried creating one container using config.json containing customized env variable `TEST=TEST`
![image](https://user-images.githubusercontent.com/2051711/139844480-d2954046-d688-46c3-abef-50491cb8fcec.png)

Then run `runc exec --env TEST2=TEST2 container_id /bin/bash` to start the same container, then the newly added env variable is also shown inside the container?
![image](https://user-images.githubusercontent.com/2051711/139844408-0febf997-b616-4126-9e15-85b8e2b1c79f.png)

**Comment by kocho1984** [comment]
I run container in a slightly different way:
`runc exec --process config.json --env TEST2=TEST2 CONTAINER_ID`
I use both `--process` and `--env` parameters.

This is part of my config.json
```
"env": [
    "TEST=test"
],
```

**Comment by JasonYangShadow** [comment]
btw, how did you run it? I can not reproduce your scenario
![image](https://user-images.githubusercontent.com/2051711/139854210-73fd1454-0055-465a-8814-4a1b58b886ef.png)

```
runc version 1.0.2
commit: v1.0.2-0-g52b36a2
spec: 1.0.2-dev
go: go1.16.8
libseccomp: 2.3.1
```
beside that I think when `process` arg is set runc will directly load settings from json file rather than processing the left args such as `env`.  Maybe you can directly edit the config.json file.

**Comment by kocho1984** [comment]
I have also `"cwd": "/",` in config.json.
In config.json, env variables must be hardcoded. That's why I would like to use `--env `parameter as well. Please imagine following use case. I would like to set HOSTNAME variable depending on the output from `hostname` command:
![image](https://user-images.githubusercontent.com/33716508/139856044-a373595b-a106-4c1c-9b70-e46ddb84bf28.png)

With `--env`, you can set value of env variables dynamically. In config.json you cannot do it.
I think `--env` shouldn't be ignored if we use `--process`.

**Comment by JasonYangShadow** [comment]
yes, I get your point. as all fields inside json are hardcoded. it is inflexible when executing existing container with new env settings. 

The weird thing is that I have `cwd` set inside my json file, but runc still complains about the `cwd`. Not sure about the reason.
![image](https://user-images.githubusercontent.com/2051711/140024200-19919726-50be-4326-a251-d823db4a145a.png)

**Comment by JasonYangShadow** [comment]
I think it is caused by this line?
https://github.com/opencontainers/runc/blob/a7ccc02fd50a488703beea6d2d5a33de1adc6ce1/exec.go#L185
